client/express-checkout/index.js

```diff
@@ -250,9 +255,47 @@ jQuery( ( $ ) => {
 			wcpayECE.showButton( eceButton );
 
 			eceButton.on( 'click', function ( event ) {
-				// TODO: handle cases where we need login confirmation.
+				// If login is required for checkout, display redirect confirmation dialog.
+				if ( getExpressCheckoutData( 'login_confirmation' ) ) {
+					displayLoginConfirmation( event.expressPaymentType );
+					return;
+				}
 
 				if ( getExpressCheckoutData( 'is_product_page' ) ) {
+					const addToCartButton = $( '.single_add_to_cart_button' );
+
+					// First check if product can be added to cart.
+					if ( addToCartButton.is( '.disabled' ) ) {
+						if (
+							addToCartButton.is( '.wc-variation-is-unavailable' )
+						) {
+							window.alert(
+								window?.wc_add_to_cart_variation_params
+									?.i18n_unavailable_text ||
+									__(
+										'Sorry, this product is unavailable. Please choose a different combination.',
+										'woocommerce-payments'
+									)
+							);
+						} else {
+							window.alert(
+								__(
+									'Please select your product options before proceeding.',
+									'woocommerce-payments'
+								)
+							);
+						}
+						return;
+					}
+
+					// TODO: This scenario.
+					// For PRBs wcPayECEError (paymentRequestError) is never re-assigned so it might be not working properly.
+					if ( wcPayECEError.length > 0 ) {
+						window.alert( wcPayECEError );
```

Comment:
>Should we create an issue to replace this and above window alerts with a built-in WooCommerce error? Displaying errors like that is not the best user experience we can offer

Currently, this is how PRBs function. While I agree that improvements can be made, it would be best to address those in a separate issue. What do you think, @bborman22? Is it worth exploring this further?


>especially wcPayECEError being an array that can potentially carry multiple errors.

I've already updated the code to treat `wcPayECEError` as string but improved it in 
https://github.com/Automattic/woocommerce-payments/pull/8987/commits/8ebe1adf70249aabaf599fd87e65f8f4e6c8eb25

Reply:
I would agree that we should fix this in a separate issue. We should create one and make sure it has the focus of checkout payments and type of enhancement so it gets triaged properly.

Reply:
@asumaran, could you create an issue for this?

Reply:
Issue created https://github.com/Automattic/woocommerce-payments/issues/9069